### DISCUSSION TITLE
Constant Kernel in GPyTorch

### DISCUSSION THREAD

**Discussion by GHU2021** [cornellius-gp/gpytorch#1623] [Q&A] [answered]
Hi,

Constant kernel is defined in Scikit-learn, Gpflow, and other Python libraries but it is not defined in GPyTorch.
Using constant kernel in some cases can improve the performance of Gaussian processes.
How can we define it?

Thank you.

**Comment by Balandat** [comment, marked as answer]
You need to subclass the `Kernel` class and implement the behavior you need in the `forward` method. Probably a good place to start to get some inspiration is the `ScaleKernel` implementation.

**Reply by GHU2021** [reply]
Thank you very much.

**Comment by AndrewFalkowski** [comment]
@GHU2021 were you ever able to get a constant kernel up and running?
